src/pkg/projects/Microsoft.NETCore.App/Microsoft.NETCore.App.targets

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <Target Name="EnsureNETCoreAppRuntime" 
+          Condition="'$(RuntimeIdentifier)' != '' AND '$(EnsureNETCoreAppRuntime)' != 'false'" 
+          AfterTargets="RunResolvePackageDependencies">
+    <ItemGroup>
+      <_netCoreAppPackageDependencies 
+          Include="@(PackageDependencies->WithMetadataValue('ParentPackage', '$(MSBuildThisFileName)/$(_Microsoft_NETCore_App_Version)'))" />
+      <_activeRIDNetCoreAppPackageDependencies 
+          Include="@(_netCoreAppPackageDependencies->WithMetadataValue('ParentTarget', '$(NuGetTargetMoniker)/$(RuntimeIdentifier)'))" />
+      <_activeTFMNetCoreAppPackageDependencies 
+          Include="@(_netCoreAppPackageDependencies->WithMetadataValue('ParentTarget', '$(NuGetTargetMoniker)'))" />
+      <_ridSpecificNetCoreAppPackageDependencies 
+          Include="@(_activeRIDNetCoreAppPackageDependencies)" 
+          Exclude="@(_activeTFMNetCoreAppPackageDependencies)" />
+    </ItemGroup>
+    
+    <Error Condition="'@(_ridSpecificNetCoreAppPackageDependencies)' == ''"
+           Text="Project is targeting runtime '$(RuntimeIdentifier)' but did not resolve any runtime-specific packages for the '$(MSBuildThisFileName)' package.  This runtime may not be supported by .NET Core." />
```

Comment:
IMO if we know 100% something is going to fail in a blocking way later but can provide a better blocking error early that is preferable.

Folks always focus on the blocking error and will often miss a warning which occurred earlier.

If we can come up with any scenario where someone would want to build with a RID and not care that they are missing the runtime package then I'd relax this to a warning but I can't come up with that.  I considered the case of RID-specific non-selfcontained, but even there we'll get the RID-specific package and trim it out later.